zombies.js

```diff
@@ -92,123 +131,197 @@
+      console.log(`${Player.name} got ${item.name}`);
+      this._pack.push(item)
+      return true
+    } else {
+      console.log('pack is full')
+      return false;
+    }
+  }
+
+  /**
+   * Player Class Method => discardItem(item)
+   * -----------------------------
+   * Player discards an item from their pack.
+   *
+   * Use Array's indexOf method to check if the pack contains the item.
+   * If an item is not found in the pack, indexOf returns -1.
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf
+   *
+   * If the item is in the pack, remove it from the pack using Array's splice method.
+   * Print the player and item names and a message saying the item was discarded.
+   * Return true for the successful discard.
+   * Note: The splice method can also be used for array element replacement.
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice
+   *
+   * If the item is not in the pack, return a message with the item name saying
+   *   nothing was discarded since the item could not be found.
+   * Return false in this case.
+   *
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name discardItem
+   * @param {Item/Weapon/Food} item   The item to discard.
+   * @return {boolean} true/false     Whether player was able to remove item from pack.
+   */
+
+  discardItem = function (item) {
+    let index = this._pack.indexOf(item);
+    if (index === -1) {
+      console.log(`${item.name} was not found`);
+    } else {
+      this._pack.splice(this._pack.indexOf(item), 1);
+      console.log(`${item.name} removed from inventory`);
+      return true;
+    }
+  }
+  /**
+   * Player Class Method => equip(itemToEquip)
+   * -----------------------------
+   * Player equips a weapon item.
+   *
+   * Player can only equip Weapon instances.
+   * Player can only equip weapon items from their pack.
+   *
+   * If the player already has a weapon equipped (the equipped property
+   *   is set to an Item), find the itemToEquip in the pack and replace
+   *   it with the currently equipped item.  Then set the equipped property
+   *   to the itemToEquip.
+   * However, if the player doesn't already have a weapon equipped, simply
+   *   equip that item and remove it from the pack.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name equip
+   * @param {Weapon} itemToEquip  The weapon item to equip.
+   */
+
+  equip = function (item) {
+
+    let isWeapon = item instanceof Weapon;
+    let index = this._pack.indexOf(item);
+
+    if (index !== -1) {
+      if (isWeapon) {
+        if (this.equipped === false) {
+          this.discardItem(item);
+          this.equipped = item;
+        } else {
+          this.discardItem(item);
+          this.takeItem(this.equipped);
+          this.equipped = item;
+        }
+      }
+      return false;
+    }
+    return false;
```

Comment:
This will always return false since you're not returning true in any of the innermost if conditionals.